src/utils/drag-and-drop-utils.js

```diff
@@ -68,6 +68,9 @@ function canDrop(dropTargetProps, monitor, isHover = false) {
 
     const targetDepth = getTargetDepth(dropTargetProps, monitor);
     const draggedNode = monitor.getItem().node;
+    const parentPath = dropTargetProps.path.slice(0, -1);
```

Comment:
@gaelduplessix 

> would you have any advice on how to reliably retrieve the expected parent node while dragging ?

This is where I got stuck on this problem as well. Without running the whole insert function, you can't see where the node will end up. That ends up being a heavy operation if you're calling it on every drag event (for canDrop). I brainstormed for a while about some shortcuts to take to just get the parent node without calling the entire insert, but nothing particularly fast came to mind.